### PR TITLE
FC-421 apprenticeships search results

### PR DESCRIPTION
**Pull request by desderry**

### CODE REVIEW

File: src/Web/Sfa.Das.Sas.Shared.Components.UnitTests/ViewComponents/Fat/FatSearchResultsViewViewComponentTests.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+using Sfa.Das.Sas.Shared.Components.Mapping;
+using Sfa.Das.Sas.Shared.Components.ViewComponents.Fat;
+using Sfa.Das.Sas.Shared.Components.ViewComponents.Fat.SearchResults;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Sfa.Das.Sas.Shared.Components.UnitTests.ViewComponents.Fat
+{
+    [TestFixture]
+    public class FatSearchResultsViewComponentTests : ViewComponentTestsBase
```
Comment (leewadhams) [inline src/Web/Sfa.Das.Sas.Shared.Components.UnitTests/ViewComponents/Fat/FatSearchResultsViewViewComponentTests.cs]: Name of class and file don't match

File: src/Web/Sfa.Das.Sas.Shared.Components.UnitTests/appsettings.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Warning"
+    }
+  },
+  "AllowedHosts": "*",
+  "fatSharedComponents": {
+     "FatApiBaseUrl": "https://findapprenticeshiptraining-api.sfa.bis.gov.uk/"
```
Comment (leewadhams) [inline src/Web/Sfa.Das.Sas.Shared.Components.UnitTests/appsettings.json]: I guess it doesn't really matter but this is the old api url.

File: src/Web/Sfa.Das.Sas.Shared.Components.Web/Views/Components/Search.cshtml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            <td class="govuk-table__cell">bool</td>
+            <td class="govuk-table__cell">False</td>
+            <td class="govuk-table__cell">Determines if the search button should be rendered inline with the keyword search input box</td>
+
+        </tr>
+    </tbody>
+</table>
+<h2 class="govuk-heading-m">Examples</h2>
+<h3 class="govuk-heading-s">Default</h3>
+<p class="govuk-body">Either when no inline parameter is provideded or set to false.</p>
```
Comment (leewadhams) [inline src/Web/Sfa.Das.Sas.Shared.Components.Web/Views/Components/Search.cshtml]: Spelling

File: src/Web/Sfa.Das.Sas.Shared.Components/Configuration/FatSharedComponentsConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (leewadhams) [inline src/Web/Sfa.Das.Sas.Shared.Components/Configuration/FatSharedComponentsConfiguration.cs]: Redundant usings

File: src/Web/Sfa.Das.Sas.Shared.Components/Configuration/FatSharedComponentsConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+using System.Collections.Generic;
+using System.Text;
+using Sfa.Das.Sas.Core.Configuration;
+
+namespace Sfa.Das.Sas.Shared.Components.Configuration
+{
+    public class FatSharedComponentsConfiguration : IFatConfigurationSettings
+    {
+        public virtual string FatApiBaseUrl { get; set; }
```
Comment (leewadhams) [inline src/Web/Sfa.Das.Sas.Shared.Components/Configuration/FatSharedComponentsConfiguration.cs]: Why virtual?

File: src/Web/Sfa.Das.Sas.Shared.Components/Controllers/FatController.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
```
Comment (leewadhams) [inline src/Web/Sfa.Das.Sas.Shared.Components/Controllers/FatController.cs]: Redundant usings

File: src/Web/Sfa.Das.Sas.Shared.Components/DependencyResolution/SharedComponentsRegistry.cs [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+using SFA.DAS.AssessmentOrgs.Api.Client;
+using SFA.DAS.NLog.Logger;
+
+namespace Sfa.Das.Sas.Shared.Components.DependencyResolution
+{
+    public static class SharedComponentsRegistry
+    {
+        public static void AddFatSharedComponents(this IServiceCollection services, IFatConfigurationSettings configuration, bool useElastic = false)
+        {
+            //   services.AddTransient<IMyService, MyService>();
```
Comment (leewadhams) [inline src/Web/Sfa.Das.Sas.Shared.Components/DependencyResolution/SharedComponentsRegistry.cs]: Commented out boilerplate code.

File: src/Web/Sfa.Das.Sas.Infrastructure/Mapping/ApprenticeshipSearchResultsItemMapping.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+                    EffectiveFrom = document.EffectiveFrom,
+                    EffectiveTo = document.EffectiveTo,
+                    Level = document.Level,
+                    Published = document.Published,
+                    PathwayName = document.PathwayName,
+                    Title = document.Title,
+                    Keywords = document.Keywords?.Any() == true ? document.Keywords.ToList() : null,
+
+                    JobRoles = document.JobRoles?.Any() == true ? document.JobRoles.ToList() : null,
+                    ApprenticeshipType = document.ProgrammeType == SFADASApprenticeshipsApiTypesV2ApprenticeshipSearchResultsItem.ProgrammeTypeEnum.NUMBER_0 ? ApprenticeshipType.Framework : ApprenticeshipType.Standard
```
[the remark this answers is not in the document]
Comment (desderry) [reply src/Web/Sfa.Das.Sas.Infrastructure/Mapping/ApprenticeshipSearchResultsItemMapping.cs]: yes, can change but will break if we re-generate the api client

File: src/Web/Sfa.Das.Sas.Shared.Components/Configuration/FatSharedComponentsConfiguration.cs (hunk 5 above)
Comment (desderry) [reply src/Web/Sfa.Das.Sas.Shared.Components/Configuration/FatSharedComponentsConfiguration.cs]: no reason whatsoever, will update

**Pull request merged**